syntax = "proto2";

package grpc.cpp.test.util;

message EchoRequest {
  optional string message = 1;
}

message EchoResponse {
  optional string message = 1;
}

service TestService {
  rpc Echo(EchoRequest) returns (EchoResponse);
  rpc RequestStream(stream EchoRequest) returns (EchoResponse);
  rpc ResponseStream(EchoRequest) returns (stream EchoResponse);
  rpc BidiStream(stream EchoRequest) returns (stream EchoResponse);
  rpc Unimplemented(EchoRequest) returns (EchoResponse) {
  }
}
